tests/ui/alloc-error/alloc-error-handler-bad-signature-1.stderr

```diff
@@ -1,16 +1,12 @@
 error[E0308]: mismatched types
-  --> $DIR/alloc-error-handler-bad-signature-1.rs:10:1
+  --> $DIR/alloc-error-handler-bad-signature-1.rs:9:1
    |
-LL |    #[alloc_error_handler]
-   |    ---------------------- in this procedural macro expansion
-LL | // fn oom(
-LL | ||     info: &Layout,
-LL | || ) -> ()
-   | ||_______- arguments to this function are incorrect
-LL | |  {
-LL | |      loop {}
-LL | |  }
-   | |__^ expected `&Layout`, found `Layout`
+LL |   #[alloc_error_handler]
+   |   ^^^^^^^^^^^^^^^^^^^^^^ expected `&Layout`, found `Layout`
+LL | / fn oom(
+LL | |     info: &Layout,
+LL | | ) -> ()
+   | |_______- arguments to this function are incorrect
```

Comment:
this lost the `in this procedural macro expansion` message

Reply:
I was personally feeling that pointing at the attribute made that label redundant. We can get it back, but don't feel too convinced we need it back :)

Reply:
well... the error just below still has it on the attribute 😆 

I'm mostly worried about the span being nonsense for a `expected Foo, found Bar` error.

Independently of this PR: why doesn't `alloc_error_handler` preserve spans and point to the actual user-written argument 🙃 ?

Reply:
Let me know what you think of the current output:

https://github.com/rust-lang/rust/pull/109082/commits/05b32a647aa74fefaa511500eeda0703a172608c

Reply:
jup, that's good!